tests/std/tests/P0040R3_parallel_memory_algorithms/test.cpp

```diff
@@ -0,0 +1,448 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <execution>
+#include <memory>
+#include <numeric>
+#include <vector>
+
+#include <parallel_algorithms_utilities.hpp>
+
+using namespace std;
+using namespace std::execution;
+
+class wrap_uchar {
+private:
+    unsigned char ch_ = 42;
+
+public:
+    bool is_expected() const noexcept {
+        return ch_ == 42;
+    }
+};
+
+static_assert(sizeof(wrap_uchar) == 1);
+
+const auto expectation = [](const wrap_uchar& x) { return x.is_expected(); };
+
+const auto expectation_zero = [](int n) { return n == 0; };
+
+constexpr auto bad_uchar = static_cast<unsigned char>(0xcd);
+constexpr auto bad_int   = static_cast<int>(0xdeadbeef);
+
+struct resetting_guard {
+    int* ptr_ = nullptr;
+
+    resetting_guard() = default;
+
+    ~resetting_guard() {
+        if (ptr_) {
+            *ptr_ = 0;
+        }
+    }
+
+    resetting_guard(const resetting_guard&)            = delete;
+    resetting_guard& operator=(const resetting_guard&) = delete;
+};
+
+template <class T>
+struct deallocating_only_deleter {
+    size_t count_ = 0;
+
+    void operator()(T* ptr) const noexcept {
+        allocator<T>{}.deallocate(ptr, count_);
+    }
+};
+
+template <class T>
+unique_ptr<T, deallocating_only_deleter<T>> make_constructed_nondestroying_buffer(size_t n) {
+    if (n == 0) {
+        return unique_ptr<T, deallocating_only_deleter<T>>{};
+    }
+
+    allocator<T> al;
+    unique_ptr<T, deallocating_only_deleter<T>> up{al.allocate(n), deallocating_only_deleter<T>{n}};
+    for (size_t i = 0; i != n; ++i) {
+        allocator_traits<allocator<T>>::construct(al, up.get() + i);
+    }

```

Comment:
No change requested: it seems unusual that an array of elements is being stored in `unique_ptr<T, DEL>` instead of `unique_ptr<T[], DEL>`. However, I am unsure of the implications of making such a change (some are positive; `unique_ptr<T[]>` disables conversions that are bogus for arrays). This is only test code, so it's allowed to be somewhat squirrelly. :chipmunk: